lib/pathfinder/PathfinderOptions.cpp

```diff
@@ -60,6 +73,7 @@ SingleHeroPathfinderConfig::~SingleHeroPathfinderConfig() = default;
 SingleHeroPathfinderConfig::SingleHeroPathfinderConfig(CPathsInfo & out, CGameState * gs, const CGHeroInstance * hero)
 	: PathfinderConfig(std::make_shared<NodeStorage>(out, hero), buildRuleSet())
 {
+	checkIfTownlessAndAllowOneWayTeleports(hero);
```

Comment:
I think this would also affect pathfinder for human player

Reply:
I tried implementing the changes via `PathfinderSettings`, as nullkiller suggested on Discord, but after making numerous changes in various places, it didn't work. This is because `SingleHeroPathfindingConfig` inherits from `PathfinderConfig`. Should I try another round on this ? I'm not an experienced C++ developer (well... not even a developer by trade, I code because I enjoy it), so if you could give me some hints, I will work in that direction to do it the right way ;)

Reply:
Yes because AI needs to know that it needs to visit monolith first to continue path. This can be achieved by linking exit node to entrance one using theNodeBefore property. See AINodeStorage::commit for Whirlpool.

Reply:
AIPath at the end should contain 2 nodes: monolith entrance, target tile

Reply:
You can use some very simple map with oneway and town behind it for testing and temporarily change
#define AI_TRACE_LEVEL 2 in AINodeStorage.h to see more logs. There in logs you should see paths printed. Path to town should contain monolith entrance tile.